crates/libcontainer/src/utils.rs

```diff
@@ -282,6 +283,39 @@ pub fn validate_spec_for_new_user_ns(spec: &Spec) -> Result<(), LibcontainerErro
     Ok(())
 }
 
+// Generic retry function with delay and policy.
+// Retries the operation `op` up to `attempts` times if it fails.
+// Waits for `delay` duration between retries.
+// Only retries if the error satisfies the `policy` function.
+pub fn retry<F, T, E, P>(
+    mut op: F,
+    attempts: u32,
+    #[cfg_attr(test, allow(unused_variables))] delay: Duration,
```

Comment:
Hey, I think it should be ok to keep the delay in tests as well. with 100 ms delay and 2 attempts after delay, each test would take at most 200 ms ; even if there are 10 tests we might have a delay of 2 sec total, I think that is ok.

Reply:
That is certainly true.
I'd like to hear @utam0k's opinion.
https://github.com/youki-dev/youki/pull/3157#discussion_r2070215467

Reply:
My concern is that if this function is used in many places, the waiting time will increase. However, since this is not the case now, it is acceptable to just leave the comment and make it const.